src/websocket_account_subscriber.rs

```
@@ -0,0 +1,138 @@
+            let pubkey = self.pubkey.clone();
+            async move {
+                let (mut account_updates, account_unsubscribe) = pubsub
+                    .account_subscribe(&pubkey, Some(account_config))
+                    .await
+                    .unwrap();
+
+                loop {
+                    tokio::select! {
+                        message = account_updates.next() => {
+                            match message {
+                                Some(message) => {
+                                    let slot = message.context.slot;
+                                    if slot >= latest_slot {
+                                        latest_slot = slot;
+                                        let account_update = AccountUpdate {
+                                            pubkey: pubkey.to_string(),
+                                            data: message.value,
+                                            slot,
+                                        };
+                                        event_emitter.emit(susbscription_name, Box::new(account_update));
+                                    }
+                                }
+                                None => {
+                                    log::warn!("{}: Account stream interrupted", susbscription_name);
+                                    account_unsubscribe().await;
+                                    break;
+                                }
+                            }
+                        }
+                        unsub = unsub_rx.recv() => {
+                            if let Some(_) = unsub {
+                                log::debug!("{}: Unsubscribing from account stream", susbscription_name);
+                                account_unsubscribe().await;
+                                break;
+
+                            }
+                        }
+                    }
+                }
+            }
```

we will need some way to indicate when this subscription task has ended or auto-reconnection, otherwise will get silent failures